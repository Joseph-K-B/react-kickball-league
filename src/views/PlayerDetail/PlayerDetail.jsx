import { useEffect, useState } from "react"
import { Link, useParams } from "react-router-dom"
import { getPlayerById } from "../../services/players"

function PlayerDetail() {
    const{ id } = useParams()
    const[player, setPlayer] = useState(null)

    useEffect(() => {
        getPlayerById(id).then((res) => setPlayer(res))
    }, [id])

    if(!player) return <h2>Loading player info...</h2>

    return(
        <>
        <h1>{player.name}</h1>
        <Link to={`/teams/${player.teams.id}`}>
        <h3>{player.teams.name}</h3>
        </Link>
        <h3>{player.teams.city}, {player.teams.state}</h3>        
        <h3>{player.position}</h3>
        </>
    )
}
export default PlayerDetail