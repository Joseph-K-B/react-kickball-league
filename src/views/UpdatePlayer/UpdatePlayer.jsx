import { useEffect, useState } from "react";
import { useHistory, useParams } from "react-router-dom";
import { Link } from "react-router-dom";
import PlayerUpdateForm from "../../components/PlayerUpdateForm";
import { getPlayerById, updatePlayerById } from "../../services/players";



function UpdatePlayer() {
    const { id } = useParams();
    const [player, setPlayer] = useState([])
    const [name, setName] = useState('');
    const [position, setPosition] = useState('');
    const history = useHistory();

    useEffect(() => {
        getPlayerById(id).then((res) => setPlayer(res))
    }, [id]);

    const handleUpdatePlayerSubmit = async (e) => {
        e.preventDefault();
        const res = await updatePlayerById( id, {name, position})
        history.push(`/players/${res[0].id}`)
    };

    if(!player) return <h2>Loading Player...</h2>

    return (
        <>
        <h1>{player.name}</h1>
        <h3>{player.position}</h3>
        <fieldset>
            <legend>Update Player</legend>
            <PlayerUpdateForm
                handleSubmit={handleUpdatePlayerSubmit}
                name={name}
                position={position}
                setName={setName}
                setPosition={setPosition}
            />
        </fieldset>
        <Link className='link' to='/players'>
            <p>Player Directory</p>
        </Link>
        </>
    )
}
export default UpdatePlayer;