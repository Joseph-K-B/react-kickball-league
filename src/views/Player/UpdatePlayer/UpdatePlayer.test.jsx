import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import {Router, Route } from 'react-router-dom';
import { createMemoryHistory } from "history";
import userEvent from '@testing-library/user-event';
import UpdatePlayer from './UpdatePlayer'
import PlayerDetail from '../PlayerDetail/PlayerDetail';
import { useEffect } from 'react';

const mockPlayer = {
    id: 12,
    created_at: '2021-12-08T16:09:33.408898+00:00',
    name: 'player name',
    position: 'benchwarmer',
    teamId: 1
}

const server = setupServer(
rest.get('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/players', 
    (req, res, ctx) => {
        return res(ctx.json(mockPlayer));
    }),
    rest.post('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/players', 
    (req, res, ctx) => {
        return res (
            ctx.json([mockPlayer]));
    })
);

beforeAll(() => {
    server.listen();
});

afterAll(() => {
    server.close();
});

it('renders form to update player', async () => {
    const history = createMemoryHistory();
    history.push('/players/1/update');

    const {container} = render(
        <Router history={history}>
            <Route>
                <UpdatePlayer />
            </Route>
        </Router>
    );

    await screen.findByText('Update Player');

    const nameField = screen.getByLabelText(/Name:/);
    const positionField = screen.getByLabelText(/Position:/);
    const submitBtn = screen.getByRole('button', {name: 'edit player'})

    userEvent.type(nameField, 'Another Player');
    userEvent.type(positionField, 'cheerleader');
    userEvent.click(submitBtn);

    expect(container).toMatchSnapshot();
    await screen.findByText('player name');
})