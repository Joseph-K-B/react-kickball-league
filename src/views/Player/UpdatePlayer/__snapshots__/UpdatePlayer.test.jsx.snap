// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form to update player 1`] = `
<div>
  <h1 />
  <h3 />
  <fieldset>
    <legend>
      Update Player
    </legend>
    <form>
      <label
        for="name"
      >
        Name:
      </label>
      <input
        id="name"
        type="text"
        value="Another Player"
      />
      <label
        for="position"
      >
        Position:
      </label>
      <input
        id="position"
        type="text"
        value="cheerleader"
      />
      <button
        aria-label="edit player"
        type="submit"
      >
        Submit
      </button>
    </form>
  </fieldset>
  <a
    class="link"
    href="/players"
  >
    <p>
      Player Directory
    </p>
  </a>
</div>
`;
